src/ser.rs

```diff
@@ -234,7 +241,7 @@ impl<'a> Serializer<'a> {
         Serializer {
             dst: dst,
             state: State::End,
-            settings: Settings::default(),
+            settings: Rc::new(Settings::default()),
```

Comment:
`Settings` cannot be a `&` because it has to be returned here (unless we wanted the user to create and pass in the settings... which was the API initially)